
syntax = "proto3";

package lemming.dataplane.sai;

import "dataplane/standalone/proto/common.proto";

option go_package = "github.com/openconfig/lemming/dataplane/standalone/proto";


enum SamplepacketAttr {
	SAMPLEPACKET_ATTR_UNSPECIFIED = 0;
	SAMPLEPACKET_ATTR_SAMPLE_RATE = 1;
	SAMPLEPACKET_ATTR_TYPE = 2;
	SAMPLEPACKET_ATTR_MODE = 3;
}

message CreateSamplepacketRequest {
	uint64 switch = 1;
	
	uint32 sample_rate = 2;
	SamplepacketType type = 3;
	SamplepacketMode mode = 4;
	
}

message CreateSamplepacketResponse {
	uint64 oid = 1;
	
	
}

message RemoveSamplepacketRequest {
	uint64 oid = 1;
	
	
}

message RemoveSamplepacketResponse {
	
	
}

message SetSamplepacketAttributeRequest {
	uint64 oid = 1;
	oneof attr {
	uint32 sample_rate = 2;
	}
}

message SetSamplepacketAttributeResponse {
	
	
}

message GetSamplepacketAttributeRequest {
	uint64 oid = 1;
	repeated SamplepacketAttr attr_type = 2;
	
	
}

message GetSamplepacketAttributeResponse {
	repeated SamplepacketAttribute attr = 1;
	
	
}


service Samplepacket {
	rpc CreateSamplepacket (CreateSamplepacketRequest) returns (CreateSamplepacketResponse) {}
	rpc RemoveSamplepacket (RemoveSamplepacketRequest) returns (RemoveSamplepacketResponse) {}
	rpc SetSamplepacketAttribute (SetSamplepacketAttributeRequest) returns (SetSamplepacketAttributeResponse) {}
	rpc GetSamplepacketAttribute (GetSamplepacketAttributeRequest) returns (GetSamplepacketAttributeResponse) {}
}
